### ISSUE TITLE
Group Publish Not Displaying Posts

### ISSUE THREAD

**Issue by custom-jonathan** [urbit/urbit#2931]
**Describe the bug**
Someone added a post to the group notebook, but doesn't show up.

**Expected behavior**
Anyone in the group adds a note, and it shows up in the notebook.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser Brave
 - `%base-hash 0vi.m33cm.0pvjs.j53ra.06tn4.klqk2.tjlo8.ev4pa.qgr2l.lecp5.0hgjj`

**Additional context**
I navigated to the notebook contents via dojo and the post was there in the directory, but nobody but the author could see it in Landscape. I tried running `|commit %home` and `:goad %force` but neither seemed to do anything.

**Comment by philipcmonk** [comment]
For the author it shows up immediately regardless of whether it's gotten to the server yet.  Most likely issues are not being able to communicate or a stuck clay queue.  Maybe try |cancel %force to see if that's stuck.

**Comment by custom-jonathan** [comment]
Did `|cancel` and got a queue length of 0. Rand `|cancel %force` just to make sure, and got the same response. Checked the notebook via Landscape after a hard refresh with the note still not showing up.

**Comment by custom-jonathan** [comment]
Don't know if this helps but here are my publish hashes:
```
> .^(@uv %cz /=base=/app/publish)                                                      |
0v10.2ul4t.9g5s7.i275m.h6lg0.rh2od.0e8ti.r4nua.bd7ne.d37v1.ogkp7                       |
> .^(@uv %cz /=home=/app/publish)                                                      |
0v1l.6op5q.62gok.siml3.dri67.uo9r1.67iv5.i5d8q.lrpr2.ii30c.ht70o
```

**Comment by ixv** [comment]
That does help. I would suggest mounting both base and home, and manually copying all the files from base to home.

**Comment by ixv** [comment]
Is this notebook hosted on your ship, or someone else's ship in the group?

**Comment by custom-jonathan** [comment]
This is a notebook hosted on my ship.

I've run `|mount %` before so my pier is visible to unix. Are you saying copy & paste all the files from "base" to "home" and overwriting everything there then running `|commit %home`? Or is it some other process?

**Comment by ixv** [comment]
Yeah, thats right. Normally I would suggest doing `|merge %home our %base, =gem %that`, but that would clobber your notebooks.

Also, can you make sure you have connectivity with the ship that tried to post?

**Comment by philipcmonk** [comment]
That command is expected to differ, right?  Since the notebooks are stored in app/publish

**Comment by custom-jonathan** [comment]
`|hi`'d them with an instant success message.

**Comment by ixv** [comment]
@philipcmonk What should you run instead? I actually am not familiar with the different merge strategies aside from %that and %init.

**Comment by custom-jonathan** [comment]
(My non-group affiliated notebook has 200+ posts, so that would be a bummer to wipe them. X^D )

**Comment by philipcmonk** [comment]
It's publish.hoon that you want to check whether it's different, right?  If so, then try `.^(@uv %cz /=base=/app/publish/hoon)` and same for home and see if they're actually different.  If it's the js, then publish/js instead of publish/hoon.

Something you can do is merge into a new desk, try your merges there, and only after everything looks good, merge into %home.

**Comment by philipcmonk** [comment]
You can mount any/all of these desks and copy files directly.  For example, you can merge %init from %home to %scratch, then merge %that from %base to %scratch, then in unix copy the notebooks (in app/publish/notebooks i think) from %home to %scratch, commit that, and then merge (with no strategy, "auto") from %scratch back into %home

**Comment by custom-jonathan** [comment]
```
> .^(@uv %cz /=base=/app/publish/hoon)
0vt.62s6n.vsaup.lsp4p.lou7f.j4a7i.gglfc.thjq4.vqrhj.se446.o9beu
> .^(@uv %cz /=home=/app/publish/hoon)
0vt.62s6n.vsaup.lsp4p.lou7f.j4a7i.gglfc.thjq4.vqrhj.se446.o9beu
```

**Comment by custom-jonathan** [comment]
Figured it wouldn't do anything but I tried the silver bullet for kicks. Went through the whole thing perfectly fine, but no change in being able to see the posts.

**Comment by custom-jonathan** [comment]
Would there be anything else I should try?

**Comment by custom-jonathan** [comment]
I'm on the most recent hash now, and the posts aren't showing up still even though they're in my pier.

**Comment by ixv** [comment]
Sorry to leave you hanging @custom-jonathan, its somewhat difficult to debug this from a distance. I would try running `:publish %reset-warp`. If that doesn't work, would you be willing to let me remote link your dojo and take a look?

**Comment by custom-jonathan** [comment]
You're a wizard! What in the world was that?! Worked like a charm; thanks!

**Comment by custom-jonathan** [comment]
The old ones showed up, but the same issue happened as soon as I tried adding a note from my backup ship.

What do you need from me to poke around? Remote link sounds awesome.

**Comment by ixv** [comment]
Hm, thats weird, `:publish %reset-warp` restarts the loop that publish runs to read files from clay, if it worked once and then failed immediately after then there's probably multiple issues here.

Did you happen to see any stack traces on either the host or your backup ship? I would try running  `|cancel` and then that `:publish %reset-warp` command again, and trying to post again, in case it was somehow a spurious error.

If still no luck, running `:dojo|allow-remote-login ~novlud-padtyv` will let me muck around on your ship, and `:dojo|revoke-remote-login ~novlud-padtyv` will revoke my access.

**Comment by custom-jonathan** [comment]
I think it might be fixed? But there was a sync in the cue and I saw a couple publish things. Cleared them out, reset my warp, and tried a new post with a backup ship that went through. But I still opened the city gates for you to poke around just in case you see something I don't notice.
